### ISSUE TITLE
Material-based timestepping enhancements

### ISSUE THREAD

**Issue by bwspenc** [idaholab/moose#13250]
## Reason
We have an existing capability to limit the time step based on material behavior in tensor_mechanics, but there are a few limitations/issues to correct, listed under Design.

## Design
1. If ComputeMultipleInelasticStress is used with no inelastic models, the material time step is 0, when it should be infinity.
2. The postprocessor that computes the allowable time step should (optionally) throw an exception if the current time step exceeds the allowable time step, so that we reject steps that significantly exceed the material time step.
3. The creep/plasticity models should have an option to limit the material time step based on the elastic strain increment, rather than just the inelastic strain increment, as is the current behavior.
4. The name of the `postprocessor_dtlim` parameter in IterationAdaptiveDT is non-intuitive. I would suggest changing it to something like `timestep_limiting_postprocessor` to be consistent with the `timestep_limiting_function` parameter that was already in that block.

## Impact
1. Change limited to that material model. This would only make models that currently don't run work.
2. Change limited to the postprocessor. We may want to change the default behavior to make it more conservative.
3. Just another option added to a material model.
4. Add a new parameter name and deprecate the old one.

**Comment by bwspenc** [comment]
FYI @acasagran

**Comment by permcody** [comment]
Wish list: Add a parameter or an option that makes the function DT behavior actually look at the functional values for limiting DT instead of just the domain. The current function limiting is already really non-intuitive in that it only pays attention to the function domain. I created a derivative version of this TimeStepper in the "Crab" repo because that behavior is currently missing.

**Comment by bwspenc** [comment]
@permcody regarding your wish list item: are you asking for an option to have a function strictly limit the size of the time step? By design, IterationAdaptiveDT allows the time step to drift away from that specified by the function, so we don't want the function to strictly drive the time step. I could see an argument for having a time-dependent limit on the time step.

**Comment by permcody** [comment]
Yes - a time-dependent limiter. While this rarely makes sense in our fully-coupled physics cases, it can be very useful for loosely coupled physics (e.g. BISON-Trace).
